### ISSUE TITLE
Update design guidelines for accessibility

### ISSUE THREAD

**Issue by PVince81** [nextcloud/nextcloud-vue#2505]
In case we don't have these yet:
    * Difference between when to use a button or a link:
      * When you open a modal or something on the same page, it should be a button.
      * When you open a new page, it should be a link. All links need to have an href attribute, and not use tabindex="0"
      * Prefer to use native HTML elements instead of aria
      * We should not use just `<div role="button">` because then we would have to implement basics like Enter and Space key functionality ourselves
    * Difference between `aria-label` and `aria-describedby`
      * aria-label is for the actual label
      * aria-describedby is for additional description
    * aria=labelledby can be used to link different elements.
    * For images, ideally always use native `img` elements, or SVG
